capstone_project/src/main/java/com/google/sps/data/PostService.java

```diff
@@ -115,4 +133,67 @@ public long upvotePost(HttpServletRequest request) {
     datastore.put(postToUpvote);
     return currentUpvotes;
   }
+
+  /**
+   * Sorts a list of entities based on the sort type passed in and returns the sorted list. If the
+   * sort type does not exist, an empty list is returned.
+   */
+  public List<Entity> sortEntities(String sortType, List<Entity> entities) {
+    Map<String, Comparator<Entity>> sorters = initializeSorters();
+    Comparator<Entity> sortMethod = sorters.get(sortType);
+    if (sortMethod == null) {
+      return Arrays.asList();
+    }
+
+    Collections.sort(entities, sortMethod);
+    return entities;
+  }
+
+  /** Returns a Map of all comparators. */
+  private Map<String, Comparator<Entity>> initializeSorters() {
+    Map<String, Comparator<Entity>> sorters = new HashMap<>();
+    sorters.put("new", sortByNew());
+    sorters.put("top", sortByTop());
+    sorters.put("trending", sortByTrending());
+    return sorters;
+  }
+
+  /** Returns a Comparator that sorts entities by new, highest timestamp to smallest timestamp. */
+  private Comparator<Entity> sortByNew() {
```

Comment:
Here and other places could be static if it doesn't rely on the fields of the class and will be the same on each call.